TestProjects/AlembicImporter/Assets/Tests/Runtime/UnitTests/AlembicExportTest.cs

```diff
@@ -109,21 +104,31 @@ public class AlembicTestBase {
             if (string.IsNullOrEmpty (_testDirectory)) {
                 return;
             }
-
+#if UNITY_EDITOR
             // Delete the directory on the next editor update.  Otherwise,
             // prefabs don't get deleted and the directory delete fails.
             EditorApplication.update += DeleteOnNextUpdate;
+#else
+            // not in editor so delete immediately
+            try {
+                Directory.Delete (filePath, recursive : true);
+            } catch (IOException) {
+                // ignore -- something else must have deleted this.
+            }
+#endif
         }
 
         // Helper for the tear-down. This is run from the editor's update loop.
         void DeleteOnNextUpdate () {
+#if UNITY_EDITOR
```

Comment:
the entire function can be within DeleteOnNextUpdate as it would be a mistake to run it in non-editor mode.

Reply:
make this function only visible in editor mode so that we get a coding error in non-editor mode.